conans/client/toolchain/meson.py

```diff
@@ -0,0 +1,179 @@
+import os
+
+from conans.client.build.cppstd_flags import cppstd_from_settings
+from conans.util.files import save
+
+import textwrap
+from jinja2 import Template
+
+
+class MesonToolchain(object):
+    _native_filename = "conan_meson_native.ini"
+    _cross_filename = "conan_meson_cross.ini"
+
+    _native_file_template = textwrap.dedent("""
+    [project options]
+    {{project_options}}
+
+    [binaries]
+    {% if c %}c = {{c}}{% endif %}
+    {% if cpp %}cpp = {{cpp}}{% endif %}
+    {% if c_ld %}c_ld = {{c_ld}}{% endif %}
+    {% if cpp_ld %}cpp_ld = {{cpp_ld}}{% endif %}
+    {% if ar %}ar = {{ar}}{% endif %}
+    {% if strip %}strip = {{strip}}{% endif %}
+    {% if as %}as = {{as}}{% endif %}
+    {% if windres %}windres = {{windres}}{% endif %}
+    {% if pkgconfig %}pkgconfig = {{pkgconfig}}{% endif %}
+
+    [built-in options]
+    {% if buildtype %}buildtype = {{buildtype}}{% endif %}
+    {% if debug %}debug = {{debug}}{% endif %}
+    {% if default_library %}default_library = {{default_library}}{% endif %}
+    {% if b_vscrt %}b_vscrt = {{b_vscrt}}{% endif %}
+    {% if b_ndebug %}b_ndebug = {{b_ndebug}}{% endif %}
+    {% if b_staticpic %}b_staticpic = {{b_staticpic}}{% endif %}
+    {% if cpp_std %}cpp_std = {{cpp_std}}{% endif %}
+    {% if c_args %}c_args = {{c_args}}{% endif %}
+    {% if c_link_args %}c_link_args = {{c_link_args}}{% endif %}
+    {% if cpp_args %}cpp_args = {{cpp_args}}{% endif %}
+    {% if cpp_link_args %}cpp_link_args = {{cpp_link_args}}{% endif %}
+    {% if pkg_config_path %}pkg_config_path = {{pkg_config_path}}{% endif %}
+    """)
+
+    def __init__(self, conanfile, env=os.environ):
+        self._conanfile = conanfile
+        self._build_type = self._conanfile.settings.get_safe("build_type")
+        self._base_compiler = self._conanfile.settings.get_safe("compiler.base") or \
+                              self._conanfile.settings.get_safe("compiler")
+        self._vscrt = self._conanfile.settings.get_safe("compiler.base.runtime") or \
+                      self._conanfile.settings.get_safe("compiler.runtime")
+        self._cppstd = cppstd_from_settings(self._conanfile.settings)
+        self._shared = self._conanfile.options.get_safe("shared")
+        self._fpic = self._conanfile.options.get_safe("fPIC")
+        self.definitions = dict()
+        self._env = env
+
+    @staticmethod
+    def _to_meson_value(value):
+        # https://mesonbuild.com/Machine-files.html#data-types
+        import six
+
+        try:
+            from collections.abc import Iterable
+        except ImportError:
```

Comment:
All this was not strictly necessary, as Meson only works with Py3, but not necessary to change.

Reply:
meson yes, but conan works with 2.x and 3.x, and they may use different interpreters (e.g. conan might use pyinstaller's one)